### PR TITLE
refactor: remove IObjectStorageFactory

### PR DESCRIPTION
**Pull request by aneojgurhem**
Remove IObjectStorageFactory since it makes a disctinction between results and payloads which is unnecessary. Moreover, we will soon introduce APIs that will need that there is no distinctions between types of data.

### CODE REVIEW

**Comment by aneojgurhem** [comment]
Tasks in a session should only access data in the same session

File: Adaptors/LocalStorage/src/ObjectStorage.cs [hunk 1]
```diff
@@ -47,16 +50,55 @@ public class ObjectStorage : IObjectStorage
...
   }
 
+  /// <inheritdoc />
+  public Task Init(CancellationToken cancellationToken)
+  {
+    _ = cancellationToken;
+    logger_.LogDebug("Initializing Local ObjectStorageFactory at path {path}, chunked by {chunkSize}",
+                     path_,
+                     chunkSize_);
+    Directory.CreateDirectory(path_);
```
Comment (lemaitre-aneo) [inline Adaptors/LocalStorage/src/ObjectStorage.cs:76]: Does this fail when the directory already exist?

File: Adaptors/S3/src/ObjectStorage.cs [hunk 2]
```diff
@@ -48,16 +52,49 @@ public class ObjectStorage : IObjectStorage
...
+  /// <inheritdoc />
+  public async Task Init(CancellationToken cancellationToken)
+  {
+    if (!isInitialized_)
+    {
+      await AmazonS3Util.DoesS3BucketExistV2Async(s3Client_,
+                                                  bucketName_);
+    }
+
+    logger_.LogInformation("ObjectStorageFactory has correctly been initialized.");
```
Comment (lemaitre-aneo) [inline Adaptors/S3/src/ObjectStorage.cs:73]: In the if ?

Comment (aneojgurhem) [reply Adaptors/S3/src/ObjectStorage.cs:73]: I did not change the implementation of the Init. I took the one from the IObjectStorageFactory

File: Adaptors/LocalStorage/src/ObjectStorage.cs (hunk 1 above)
Comment (lemaitre-aneo) [reply Adaptors/LocalStorage/src/ObjectStorage.cs:76]:
```suggestion
    // This creates all intermediate directories and does not fail if it already exists: https://learn.microsoft.com/en-us/dotnet/api/system.io.directory.createdirectory
    Directory.CreateDirectory(path_);
```

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**